controllers/eventhandlers/service.go

```diff
@@ -14,43 +13,44 @@ import (
 
 	"k8s.io/client-go/util/workqueue"
 	"sigs.k8s.io/controller-runtime/pkg/event"
-	gateway_api "sigs.k8s.io/gateway-api/apis/v1beta1"
 	mcs_api "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
 )
 
-type enqueueRequetsForServiceEvent struct {
+type enqueueRequestForServiceWithExportEvent struct {
+	log    gwlog.Logger
 	client client.Client
 }
 
-func NewEqueueRequestServiceEvent(client client.Client) handler.EventHandler {
-	return &enqueueRequetsForServiceEvent{
+func NewEqueueRequestServiceWithExportEvent(log gwlog.Logger, client client.Client) handler.EventHandler {
+	return &enqueueRequestForServiceWithExportEvent{
+		log:    log,
 		client: client,
 	}
 }
 
-func (h *enqueueRequetsForServiceEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
-	glog.V(6).Info("Event: service create")
+func (h *enqueueRequestForServiceWithExportEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	h.log.Info("Event: service create")
 	service := e.Object.(*corev1.Service)
 	h.enqueueImpactedService(queue, service)
 	h.enqueueImpactedServiceExport(queue, service)
 }
```

Comment:
Question for old code and not related to current PR,

we did this in serviceexport_controller.go:
```
svcExportEventsHandler := eventhandlers.NewEqueueRequestServiceWithExportEvent(log, r.client)

	return ctrl.NewControllerManagedBy(mgr).
		For(&mcs_api.ServiceExport{}).
		Watches(&source.Kind{Type: &corev1.Service{}}, svcExportEventsHandler).
		Complete(r)
```

if the serviceexport_controller saw a k8sService created, does it need to do`h.enqueueImpactedServiceExport(queue, service)`? That can cause one ServiceExport creation request to `Reconcile` twice?